Add codecov to the workflow

sudokoi opened the pull request.

sudokoi committed:
Add codecov to the workflow

mpeyper commented:
Just checking the log, and it's picking up more files than expected. https://github.com/OpenSourceRaidGuild/atom-pwa/pull/88/checks?check_run_id=2053227516#step:11:45

I think codecov handles merging multiple coverage files itself, so we can probable skip the step where we run `nyc` to merge them ourselves.

Also, in previous projects where I've used codecov, it would add a comment to the PR with coverage stats, which doesn't appear to be happening here. I'm not sure why bit though.

mpeyper commented:
The comment might be because of the `next` branch and the default configuration expecting `master` https://docs.codecov.io/docs/pull-request-comments#branches

mpeyper commented:
Actually, just thinking aloud, we can probably just call this action at the end of the jest and cypress jobs rather than uploading and downloading the coverage reports in a seperate job.

sudokoi committed:
Add codecov config and disable combine-coverage

Testing if codecov merges the multiple coverages by itself

sudokoi commented:
> The comment might be because of the `next` branch and the default configuration expecting `master` https://docs.codecov.io/docs/pull-request-comments#branches

I think codecov-bot needs to be enabled for this repo. Adding `codecov.yml` and specifying the branch did nothing.

sudokoi committed:
Remove codecov.yml

It is not required for specifying comment-branch. Default behavior is all branches

sudokoi commented:
> > The comment might be because of the `next` branch and the default configuration expecting `master` https://docs.codecov.io/docs/pull-request-comments#branches
> 
> I think codecov-bot needs to be enabled for this repo. Adding `codecov.yml` and specifying the branch did nothing.

Can confirm. Tested it on my fork. The bot has to be added for the pr-comments.
https://github.com/tsuki42/atom-pwa/pull/1

sudokoi commented:
> I think codecov handles merging multiple coverage files itself, so we can probable skip the step where we run `nyc` to merge them ourselves.

Commented out the `Combine coverage` step. 

>  we can probably just call this action at the end of the jest and cypress jobs rather than uploading and downloading the coverage reports in a separate job

To avoid entirely setting up the `coverage` job, and have codecov action called,  the jest and cypress jobs will probably need to be merged and then codecov action step can be called as part of the unified job.

mpeyper commented:
> To avoid entirely setting up the `coverage` job, and have codecov action called, the jest and cypress jobs will probably need to be merged and then codecov action step can be called as part of the unified job.

[According to their docs](https://docs.codecov.io/docs/merging-reports), they automatically merge reports received while the CI is pending, soy though was we can just put the action in at the end of each of the other jobs and remove the coverage job all together.

That said, I don't mind what we have now either.

mpeyper commented on .github/workflows/test.yml:
```diff
@@ -156,10 +156,8 @@ jobs:
-      - name: Combine coverage
-        run: yarn coverage:ci
+      # - name: Combine coverage
+      #   run: yarn coverage:ci
```
Remove these if not needed. Also, I think this was the only reason we were installing dependencies in this job.

sudokoi replied:
I have not cleaned the package.json scripts but have removed the entire coverage job and made the codecov step the part of other two jobs. Seems to work.

sudokoi committed:
build: remove coverage job and use codecov as a step in other two jobs
